extensions/amp-carousel/0.1/base-carousel.js

```diff
@@ -64,27 +65,20 @@ export class BaseCarousel extends AMP.BaseElement {
 
 
   buildButtons() {
-    this.prevButton_ = this.element.ownerDocument.createElement('div');
+    // TODO(aghassemi erwinm, #4146) i18n
+    this.prevButton_ = addScreenReaderButton(this.element,
+      'Previous item in carousel', () => this.interactionPrev());
     this.prevButton_.classList.add('amp-carousel-button');
     this.prevButton_.classList.add('amp-carousel-button-prev');
-    this.prevButton_.setAttribute('role', 'button');
-    // TODO(erwinm): Does label need i18n support in the future? or provide
-    // a way to be overridden.
-    this.prevButton_.setAttribute('aria-label', 'Previous item in carousel');
-    this.prevButton_.onclick = () => {
-      this.interactionPrev();
-    };
-    this.element.appendChild(this.prevButton_);
-
-    this.nextButton_ = this.element.ownerDocument.createElement('div');
+
+    this.nextButton_ = addScreenReaderButton(this.element,
+      'Next item in carousel', () => this.interactionNext());
     this.nextButton_.classList.add('amp-carousel-button');
     this.nextButton_.classList.add('amp-carousel-button-next');
-    this.nextButton_.setAttribute('role', 'button');
-    this.nextButton_.setAttribute('aria-label', 'Next item in carousel');
-    this.nextButton_.onclick = () => {
-      this.interactionNext();
-    };
-    this.element.appendChild(this.nextButton_);
+
+    // Remove the a11y class so the buttons are visible for the hint
+    this.prevButton_.classList.remove('i-amphtml-screen-reader');
```

Comment:
This is internal to `addScreenReaderButton`, so it should be done in that module.